docs/src/content/docs/reference/engines.md

```diff
@@ -50,6 +50,37 @@ gh secret set GH_AW_GITHUB_TOKEN -a actions --body "<your-github-pat>"
 The Copilot engine does not have built-in `web-search` support. You can add web search capabilities using third-party MCP servers. See the [Using Web Search](/gh-aw/guides/web-search/) for available options and setup instructions.
 :::
 
+#### Network Firewall (AWF)
+
+The Copilot engine supports an optional network firewall feature using AWF (Agent Workflow Firewall) for network egress control with domain allowlisting. AWF is sourced from [github.com/githubnext/gh-aw-firewall](https://github.com/githubnext/gh-aw-firewall).
+
+Enable the firewall feature in your workflow frontmatter:
+
+```yaml
+features:
+  firewall: true
+
+network:
+  allowed:
+    - defaults
+    - "api.example.com"
+```
+
+When enabled, AWF wraps the Copilot CLI execution and enforces domain-based network access controls, logging all network activity. This provides an additional layer of security for workflows that need strict network access control.
+
+**Firewall Configuration:**
+
+```yaml
+engine:
+  id: copilot
+  firewall:
```

Comment:
@Mossaka the firewall configuration belongs under networks since we will have unified support for it. Merge this PR and spawn a task to update the schema.